code/game/machinery/computer/camera_advanced.dm

```diff
@@ -79,14 +79,15 @@
 		eyeobj.visible_icon = TRUE
 		eyeobj.invisibility = INVISIBILITY_OBSERVER
 		if(current_user) // indent is correct: do not transfer ghosts unless it's revealed
-			current_user.transfer_observers_to(eyeobj)
+			current_user.transfer_observers_to(eyeobj, temporary = TRUE)
 
 /obj/machinery/computer/camera_advanced/proc/ConcealCameraMob()
 	if(reveal_camera_mob && eyeobj)
 		eyeobj.visible_icon = FALSE
 		eyeobj.invisibility = INVISIBILITY_ABSTRACT
 	if(current_user && eyeobj) // indent is correct: transfer ghosts when nobody uses
-		eyeobj.transfer_observers_to(current_user)
+		eyeobj.return_observers() // send ghosts back to their original orbit
+		eyeobj.transfer_observers_to(current_user) // if a ghost started observing an eye at first, the return proc won't work.
```

Comment:
Having 2 sets of logic to do something that never really makes sense to be seperate seems a little bit unnecessary, is the return_observers() actually necessary since all observers will be transfered anyway by `transfer_observers_to`.

Reply:
This is to handle a case where multiple ghosts from different orbit source being taken to a camera eye.
* A/B/C ghosts from mob `Angella`
* D/E ghosts from mob `John`
* F/G ghosts from mob `Belle`
* Angella triggered the advanced camera. ABC ghosts are now orbiting the camera eye
* John touches the camera console. DE ghosts are transfered to the camera eye
* Belle touches the camera console. FG ghosts are transfered to the camera eye
* John stops observing the camera. DE ghosts orbiting the camera eye are take back to John
* Angella stops controlling the camera console
  * Ghosts of Belle (FG) are taken back to Belle
  * Angella's ghost (ABC) are taken back to Angella too

If there's no code having this, all ghosts orbiting ABCDEFG are taken to John becasue he stopped observing. or if Angella stops controlling the camera, all ghosts are taken back to Angella, but they were originally orbiting their own before Angella.